x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/local/LocalBulk.java

```diff
@@ -52,13 +52,10 @@
     }
 
     @Override
-    public void doAdd(Collection<MonitoringDoc> docs) throws ExportException {
+    protected void doAdd(Collection<MonitoringDoc> docs) throws ExportException {
         ExportException exception = null;
 
         for (MonitoringDoc doc : docs) {
```

Comment:
Do we still need a isClosed() check here ?  (now against the AtomicBoolean)

Reply:
We do not. I removed code that cheated and called the implementations directly, which makes this whole method guarded by `add`. Also, the whole `add-flush` flow was always synchronous so we never really made use of the inner-loop escape.